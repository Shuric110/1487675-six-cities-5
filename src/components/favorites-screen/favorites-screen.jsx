import React from "react";
import {Link} from 'react-router-dom';
import PropTypes from "prop-types";

import {favoritePropType} from "../../props";
import {OFFER_TYPE_TITLES} from "../../const";
import {ratingToPercent} from "../../util";
import MainHeader from "../main-header/main-header";
import BookmarkButton from "../bookmark-button/bookmark-button";
import withAllFavorites from "../../hocs/with-all-favorites/with-all-favorites";

const FavoritesScreen = (props) => {
  const {favorites} = props;

  return (
    <div className="page">
      <MainHeader />

      <main className="page__main page__main--favorites">
        <div className="page__favorites-container container">
          <section className="favorites">
            <h1 className="favorites__title">Saved listing</h1>
            {favorites === null ? null :
              <ul className="favorites__list">
                {
                  favorites.length === 0 ?
                    (
                      <section className="favorites favorites--empty">
                        <h1 className="visually-hidden">Favorites (empty)</h1>
                        <div className="favorites__status-wrapper">
                          <b className="favorites__status">Nothing yet saved.</b>
                          <p className="favorites__status-description">Save properties to narrow down search or plan yor future trips.</p>
                        </div>
                      </section>
                    )
                    :
                    favorites.map(({city, offers}) => (
                      <li className="favorites__locations-items" key={city}>
                        <div className="favorites__locations locations locations--current">
                          <div className="locations__item">
                            <a className="locations__item-link" href="#">
                              <span>{city}</span>
                            </a>
                          </div>
                        </div>

                        <div className="favorites__places">
                          {offers.map((offer) => {
                            const {id, placePicture, isPremium, nightlyCost, title, type, rating} = offer;

                            return (
                              <article className="favorites__card place-card" key={id}>
                                {isPremium ? (
                                  <div className="place-card__mark">
                                    <span>Premium</span>
                                  </div>
                                ) : null}
                                <div className="favorites__image-wrapper place-card__image-wrapper">
                                  <Link to={`/offer/${id}`}>
                                    <img className="place-card__image" src={placePicture} width="150" height="110" alt="Place image" />
                                  </Link>
                                </div>
                                <div className="favorites__card-info place-card__info">
                                  <div className="place-card__price-wrapper">
                                    <div className="place-card__price">
                                      <b className="place-card__price-value">&euro;{nightlyCost}</b>
                                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                                    </div>
                                    <BookmarkButton baseClassName="place-card__bookmark-button" offer={offer}>
                                      <svg className="place-card__bookmark-icon" width="18" height="19">
                                        <use xlinkHref="#icon-bookmark"></use>
                                      </svg>
                                      <span className="visually-hidden">In bookmarks</span>
                                    </BookmarkButton>
                                  </div>
                                  <div className="place-card__rating rating">
                                    <div className="place-card__stars rating__stars">
                                      <span style={{width: `${ratingToPercent(rating)}%`}}></span>
                                      <span className="visually-hidden">Rating</span>
                                    </div>
                                  </div>
                                  <h2 className="place-card__name">
                                    <a href="#">{title}</a>
                                  </h2>
                                  <p className="place-card__type">{OFFER_TYPE_TITLES[type]}</p>
                                </div>
                              </article>
                            );
                          })}
                        </div>
                      </li>
                    ))
                }
              </ul>
            }
          </section>
        </div>
      </main>
      <footer className="footer container">
        <a className="footer__logo-link" href="main.html">
          <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width="64" height="33" />
        </a>
      </footer>
    </div>
  );
};

FavoritesScreen.propTypes = {
  favorites: PropTypes.arrayOf(favoritePropType.isRequired),
};

export {FavoritesScreen};
export default withAllFavorites(FavoritesScreen);
