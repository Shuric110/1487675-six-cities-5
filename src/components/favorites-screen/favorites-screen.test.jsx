import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter as Router} from 'react-router-dom';

import {FavoritesScreen} from "./favorites-screen";

const cities = [
  {
    id: 1,
    name: `Paris`,
    coordinates: {
      latitude: 48.8606146,
      longitude: 2.3354553
    },
    zoom: 12,
  },
  {
    id: 2,
    name: `Amsterdam`,
    coordinates: {
      latitude: 48.8606146,
      longitude: 2.3354553
    },
    zoom: 12,
  },
];

const offers = [
  {
    id: 1,
    city: cities[0],
    pictures: [
      `/img/room.jpg`,
    ],
    placePicture: `/img/room.jpg`,
    coordinates: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198
    },
    mapZoom: 12,
    isPremium: false,
    isFavorite: true,
    nightlyCost: 80,
    title: `Wood and stone place`,
    type: `room`,
    rating: 4.2,
    description: `Excellent choice for travelers, offering many helpful amenities designed to enhance your stay.`,
    bedrooms: 1,
    maxAdults: 2,
    features: [
      `Wi-Fi`,
    ],
    host: {
      name: `Angelina`,
      avatar: `/img/avatar-angelina.jpg`,
      isSuper: true
    },
  },

  {
    id: 2,
    city: cities[0],
    pictures: [
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
    ],
    placePicture: `/img/studio-01.jpg`,
    coordinates: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198
    },
    mapZoom: 12,
    isPremium: true,
    isFavorite: true,
    nightlyCost: 80,
    title: `Wood and stone place`,
    type: `room`,
    rating: 4.2,
    description: `Excellent choice for travelers, offering many helpful amenities designed to enhance your stay.`,
    bedrooms: 1,
    maxAdults: 2,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Cable TV`,
      `Fridge`
    ],
    host: {
      name: `Angelina`,
      avatar: `/img/avatar-angelina.jpg`,
      isSuper: true
    },
  },

  {
    id: 3,
    city: cities[1],
    pictures: [
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/apartment-01.jpg`,
    ],
    placePicture: `/img/apartment-01.jpg`,
    coordinates: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
    mapZoom: 12,
    isPremium: false,
    isFavorite: true,
    nightlyCost: 132,
    title: `Canal View Prinsengracht`,
    type: `apartment`,
    rating: 4,
    description: `For travelers visiting Amsterdam, Canal View Prinsengracht is an excellent choice for rest and rejuvenation.
                  Well-known for its family-friendly environment and proximity to great restaurants and attractions, Canal View Prinsengracht makes it easy to enjoy the best of Amsterdam.`,
    bedrooms: 3,
    maxAdults: 5,
    features: [
      `Washing machine`,
      `Towels`,
      `Heating`,
    ],
    host: {
      name: `Max`,
      avatar: `/img/avatar-max.jpg`,
      isSuper: false
    },
  },
];

const favorites = [
  {
    city: cities[0].name,
    offers: [
      offers[0],
      offers[1],
    ]
  },
  {
    city: cities[1].name,
    offers: [
      offers[2],
    ]
  }
];

jest.mock(`../main-header/main-header`, () => `MainHeader`);
jest.mock(`../bookmark-button/bookmark-button`, () => `BookmarkButton`);

describe(`Render FavoritesScreen`, () => {
  it(`FavoritesScreen for non-empty list renders correctly`, () => {
    const tree = renderer
      .create(
          <Router>
            <FavoritesScreen
              favorites={favorites}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`FavoritesScreen for empty list renders correctly`, () => {
    const tree = renderer
      .create(
          <Router>
            <FavoritesScreen
              favorites={[]}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
