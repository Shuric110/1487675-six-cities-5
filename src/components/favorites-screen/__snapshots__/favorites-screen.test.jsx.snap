// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FavoritesScreen FavoritesScreen for empty list renders correctly 1`] = `
<div
  className="page"
>
  <MainHeader />
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <section
            className="favorites favorites--empty"
          >
            <h1
              className="visually-hidden"
            >
              Favorites (empty)
            </h1>
            <div
              className="favorites__status-wrapper"
            >
              <b
                className="favorites__status"
              >
                Nothing yet saved.
              </b>
              <p
                className="favorites__status-description"
              >
                Save properties to narrow down search or plan yor future trips.
              </p>
            </div>
          </section>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;

exports[`Render FavoritesScreen FavoritesScreen for non-empty list renders correctly 1`] = `
<div
  className="page"
>
  <MainHeader />
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Paris
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/1"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="110"
                      src="/img/room.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        80
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <BookmarkButton
                      baseClassName="place-card__bookmark-button"
                      offer={
                        Object {
                          "bedrooms": 1,
                          "city": Object {
                            "coordinates": Object {
                              "latitude": 48.8606146,
                              "longitude": 2.3354553,
                            },
                            "id": 1,
                            "name": "Paris",
                            "zoom": 12,
                          },
                          "coordinates": Object {
                            "latitude": 52.369553943508,
                            "longitude": 4.85309666406198,
                          },
                          "description": "Excellent choice for travelers, offering many helpful amenities designed to enhance your stay.",
                          "features": Array [
                            "Wi-Fi",
                          ],
                          "host": Object {
                            "avatar": "/img/avatar-angelina.jpg",
                            "isSuper": true,
                            "name": "Angelina",
                          },
                          "id": 1,
                          "isFavorite": true,
                          "isPremium": false,
                          "mapZoom": 12,
                          "maxAdults": 2,
                          "nightlyCost": 80,
                          "pictures": Array [
                            "/img/room.jpg",
                          ],
                          "placePicture": "/img/room.jpg",
                          "rating": 4.2,
                          "title": "Wood and stone place",
                          "type": "room",
                        }
                      }
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </BookmarkButton>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="#"
                    >
                      Wood and stone place
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Private Room
                  </p>
                </div>
              </article>
              <article
                className="favorites__card place-card"
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/2"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="110"
                      src="/img/studio-01.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        80
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <BookmarkButton
                      baseClassName="place-card__bookmark-button"
                      offer={
                        Object {
                          "bedrooms": 1,
                          "city": Object {
                            "coordinates": Object {
                              "latitude": 48.8606146,
                              "longitude": 2.3354553,
                            },
                            "id": 1,
                            "name": "Paris",
                            "zoom": 12,
                          },
                          "coordinates": Object {
                            "latitude": 52.369553943508,
                            "longitude": 4.85309666406198,
                          },
                          "description": "Excellent choice for travelers, offering many helpful amenities designed to enhance your stay.",
                          "features": Array [
                            "Wi-Fi",
                            "Washing machine",
                            "Cable TV",
                            "Fridge",
                          ],
                          "host": Object {
                            "avatar": "/img/avatar-angelina.jpg",
                            "isSuper": true,
                            "name": "Angelina",
                          },
                          "id": 2,
                          "isFavorite": true,
                          "isPremium": true,
                          "mapZoom": 12,
                          "maxAdults": 2,
                          "nightlyCost": 80,
                          "pictures": Array [
                            "/img/apartment-01.jpg",
                            "/img/apartment-02.jpg",
                          ],
                          "placePicture": "/img/studio-01.jpg",
                          "rating": 4.2,
                          "title": "Wood and stone place",
                          "type": "room",
                        }
                      }
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </BookmarkButton>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="#"
                    >
                      Wood and stone place
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Private Room
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Amsterdam
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/3"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="110"
                      src="/img/apartment-01.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        132
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <BookmarkButton
                      baseClassName="place-card__bookmark-button"
                      offer={
                        Object {
                          "bedrooms": 3,
                          "city": Object {
                            "coordinates": Object {
                              "latitude": 48.8606146,
                              "longitude": 2.3354553,
                            },
                            "id": 2,
                            "name": "Amsterdam",
                            "zoom": 12,
                          },
                          "coordinates": Object {
                            "latitude": 52.3909553943508,
                            "longitude": 4.929309666406198,
                          },
                          "description": "For travelers visiting Amsterdam, Canal View Prinsengracht is an excellent choice for rest and rejuvenation.
                  Well-known for its family-friendly environment and proximity to great restaurants and attractions, Canal View Prinsengracht makes it easy to enjoy the best of Amsterdam.",
                          "features": Array [
                            "Washing machine",
                            "Towels",
                            "Heating",
                          ],
                          "host": Object {
                            "avatar": "/img/avatar-max.jpg",
                            "isSuper": false,
                            "name": "Max",
                          },
                          "id": 3,
                          "isFavorite": true,
                          "isPremium": false,
                          "mapZoom": 12,
                          "maxAdults": 5,
                          "nightlyCost": 132,
                          "pictures": Array [
                            "/img/apartment-03.jpg",
                            "/img/studio-01.jpg",
                            "/img/apartment-01.jpg",
                          ],
                          "placePicture": "/img/apartment-01.jpg",
                          "rating": 4,
                          "title": "Canal View Prinsengracht",
                          "type": "apartment",
                        }
                      }
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </BookmarkButton>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="#"
                    >
                      Canal View Prinsengracht
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Apartment
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
